pkg/serviceprovider/github/github.go

```diff
@@ -182,7 +182,11 @@ func (g *Github) CheckRepositoryAccess(ctx context.Context, cl client.Client, ac
 	tokens, lookupErr := g.lookup.Lookup(ctx, cl, accessCheck)
 	if lookupErr != nil {
 		lg.Error(lookupErr, "failed to lookup token for accesscheck", "accessCheck", accessCheck)
-		return status, fmt.Errorf("failed to lookip token for accesscheck %s/%s: %w", accessCheck.Namespace, accessCheck.Name, lookupErr)
+		if !publicRepo {
```

Comment:
Btw why do we even continue here if we already determined the repository was public? Do we learn any new information by doing the token lookup?

Reply:
idea was that we can find more information. But it's not actually needed. I've updated the code so it returns immediately at `if publicRepo`